### ISSUE TITLE
Forbid numeric object numbers

### ISSUE THREAD

**Issue by rdebath** [microsoft/AL#6257]
**Title**
Forbid explicit numeric object numbers

**Description**
Explicit object numbers can be used when calling `RUN` and `RUNMODAL`. Please generate a Codecop warning when this value is a literal numeric value like `80` and instead suggest something like `Codeunit:"Sales Post"`.  The autofix should provide the correct substitution for the number in the `Codeunit.Run(80, SalesHeader)` statement.

This should apply to all `ObjectType.RUN` methods, `ObjectType.RUNMODAL` methods  and Event subscriptions.

**Reason for the rule**
When object numbers are used unless, like me :grin:, you have perfect recall of these numbers from years of practice it's very unclear which objects they actually are.

In addition, renumbering objects in an extension _can_ be almost trivial (three regular expressions), use of literal object numbers can defeat that.

**Bad code sample**
Example of what bad code the rule should catch:

```   
page.run(130, SalesShptHdr);
 ```

**Good code sample**
Example of what code should look like:

``` 
Page.Run(Page::"Posted Sales Shipment", SalesShptHdr);
 ```

**Comment by bjarkihall** [comment]
Yes, that would be great, as long as it's a CodeCop and not required, since these should also be able to accept an integer since some patterns depend on that feature for dynamic Codeunit running.

**Comment by dtkb** [comment]
Yes, but then that would be a variable `Integer`, not a constant/literal, which is the subject of the OP.

**Comment by bjarkihall** [comment]
Ah, didn't notice that, then it makes total sense. :)

**Comment by SBalslev** [comment]
Hi! Thank you for reporting this — we appreciate you taking the time to document the issue with a clear repro.

We're currently not investing in CodeCop rule improvements, but we have this on our radar for a future effort. In the meantime, workarounds such as pragma directives or suppression attributes can be used to manage false positives.

We're closing this to keep the backlog focused, but your input is noted internally. If this becomes a significant blocker on a future version, please don't hesitate to open a new issue.

Thank you for helping us improve the AL developer experience!
